File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyPrice = prices[0];
        int maxProfit = 0;

        for (int p : prices){
            if (p < buyPrice){
                buyPrice = p;
            }

            maxProfit = Math.max(maxProfit, p - buyPrice);
        }
        return maxProfit;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy_price = prices[0]
        max_profit = 0

        for new_price in prices[1:]:
            if new_price < buy_price:
                buy_price = new_price
            
            max_profit = max(max_profit, new_price-buy_price)

        return max_profit